x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/changepoint/ChangePointDetector.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.aggs.changepoint;
+
+import org.apache.commons.math3.exception.NotStrictlyPositiveException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.xpack.ml.aggs.MlAggsHelper;
+
+/**
+ * Detects whether a series of values has a change point, by running both
+ * ChangeDetector and SpikeAndDipDetector on it. This is the main entrypoint
+ * of change point detection.
+ */
+public class ChangePointDetector {
```

Comment:
I know it is in keeping with the aggregation name, but I think we can do better on the naming. For me something like point of interest is better.
```suggestion
public class PointOfInterestDetector {
```

Reply:
Not sure if "point of interest" is that much better. E.g. it doesn't capture (non-)stationary that well.

Given that the agg is called "Change Point Aggregation" right now, I'd rather keep it like this. If we decide the ES|QL variant will be called differently (suggestions are welcome!), we can always rename it then.